placeholders.go

```diff
@@ -61,7 +61,7 @@ func parsePlaceholders(input string, r *http.Request) (string, error) {
 			if n < 1 {
 				return "", fmt.Errorf("{label0} is not supported")
 			}
-			labels := strings.Split(r.URL.Hostname(), ".")
+			labels := strings.Split(r.Host, ".")
```

Comment:
r.Host also needs the port code as we use the request's Host, which still can have a port.